### PR TITLE
Move core modules to separate enclone_ranger repo.

### PR DESCRIPTION
**Pull request by adam-azarchs**
Specifically, move:

- enclone
- enclone_args
- enclone_core
- enclone_print
- enclone_proto
- enclone_ranger
- enclone_stuff
- enclone_vars

This allows us to enforce stricter standards for the components on which
cellranger has direct dependencies, while preserving freedom for rapid
iteration in this repository.

enclone_ranger is a fork.  While changes have been made there to remove
stuff which is not required to be there, for the moment this repository
is depending on the commit from that repository which is the commit
before this one.

### CODE REVIEW

File: enclone_exec/tests/testlist.rs [hunk 1]
```diff
@@ -2,22 +2,6 @@
...
-pub fn enclone_testdata_public_bcr_human() -> String {
-    include_str!["testdata.public.bcr.human"].to_string()
-}
-pub fn enclone_testdata_public_tcr_human() -> String {
-    include_str!["testdata.public.tcr.human"].to_string()
-}
-pub fn enclone_testdata_public_tcr_mouse() -> String {
-    include_str!["testdata.public.tcr.mouse"].to_string()
-}
-pub fn enclone_testdata_public_gex_human() -> String {
```
Comment (nmmsv10x) [inline enclone_exec/tests/testlist.rs:17]: are we putting these elsewhere in the code? could not find them.

Comment (adam-azarchs) [reply enclone_exec/tests/testlist.rs:17]: Why?  This file isn't even getting compiled, let alone called by anything.  It's dead code.  Dead code should be deleted.  If we ever want it back, that's what git history is for.

Comment (nmmsv10x) [reply enclone_exec/tests/testlist.rs:17]: ah I didn't know, sounds good!

**Review by sreenathkrishnan — APPROVED** [review]
Makes sense!

**Comment by adam-azarchs** [comment]
The jenkins test is failing because it doesn't know the updated location for `enclone.test`.  We can ignore that and get it fixed after this merges.

**Comment by adam-azarchs** [comment]
There are not really any actual code changes involved here, since we're pointing at the current HEAD commit in the enclone_ranger repo.  Just a few little changes where source files were moved around a little.  This repo has an extremely non-standard CI, but I'm going to merge this as soon as the non-jenkins tests pass.

**Pull request merged**